libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -55,14 +55,15 @@ const AP_Param::GroupInfo AP_Scheduler::var_info[] = {
 };
 
 // constructor
-AP_Scheduler::AP_Scheduler(void)
+AP_Scheduler::AP_Scheduler()
 {
     AP_Param::setup_object_defaults(this, var_info);
 
     // only allow 50 to 2000 Hz
     if (_loop_rate_hz < 50) {
         _loop_rate_hz.set(50);
-    } else if (_loop_rate_hz > 2000) {
+    } 
```

Comment:
This is AFAIK slower.

Reply:
No, the compiler creates the same code. But it looks better.

Reply:
not the same in the case _loop_rate_hz < 50 as it will always do the second if, but it is negligeable

Reply:
Cannot find a difference
https://godbolt.org/g/7mrggk

Reply:
interesting assumtion from the compiler ! but on pure logic it shouldn't be the same !

Reply:
No, if the first check is false, then _loop_rate_hz is always bigger >=50. 
Then we have the other if which only checks for >2000. 
The else is just useless :/

Reply:
but if _loop_rate_hz < 50, we set it to 50, so we don't need to recheck if it is > 2000

Reply:
I would say revert it please

Reply:
> but if _loop_rate_hz < 50, we set it to 50, so we don't need to recheck if it is > 2000
if you dont want to do another check, then there should be inserted a return.

Reply:
> Cannot find a difference
https://godbolt.org/g/7mrggk

@dgrat This is not the same. Remove the `return` and the code isn't the same anymore.